servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/FilterableClientToClient.java

```diff
@@ -159,7 +159,7 @@ public StreamingHttpRequest newRequest(final HttpRequestMethod method, final Str
 
     @Override
     public HttpExecutionContext executionContext() {
-        return client.executionContext();
+        return executionContext;
```

Comment:
This is the critical difference. The returned execution context contains the correct execution strategy.